src/Agent.Worker/AsyncCommandContext.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    [ServiceLocator(Default = typeof(AsyncCommandContext))]
+    public interface IAsyncCommandContext : IAgentService
+    {
+        string Name { get; }
+        Task Task { get; set; }
+        void InitializeCommandContext(IExecutionContext context, string name);
+        void Output(string message);
+        Task WaitAsync();
+    }
+
+    public class AsyncCommandContext : AgentService, IAsyncCommandContext
```

Comment:
I like this class and how Output() is used from the command tasks. I just completed a thought exercise questioning whether this new *context* object is needed... I really like it how you designed it :) It makes a lot of sense.

I'll look over the rest of the code later, I'm pretty sure everything else is straight-forward so should go quickly.